### PR TITLE
ossec-init.conf have not correct date

### PR DESCRIPTION
**Pull request by alguacilaguamara**
Hello,

The changes made are to solve the problem of #280. The creation and initialization instructions are removed from the postinst package and passed to rules.

Regards,
Alejandro

### CODE REVIEW

**Comment by alguacilaguamara** [comment]
Hello Braulio,

I will revisit the steps to try to force the script to run as soon as possible to see if this solves the problem.

Regards,
Alejandro

**Comment by alguacilaguamara** [comment]
Hello  braulio,

I was testing if I could generate the file one point before but I only got package creation errors.

I will debug to find the point where the date is properly set.

For this I will generate another small file only with the date in the rules.

If I see that some point takes the date well I will try to pass the generation of the file with the script at that point and in the worst case I will try to create the entire file manually.

Regards.
Alejandro

**Comment by alguacilaguamara** [comment]
Hello,

I am trying to create the file before install.sh. For this I am moving the files so that they are generated properly

**Comment by alguacilaguamara** [comment]
Hello,

I am working on the manager. The agent has already found the solution, but the manager don't work with this solution.

**Comment by alguacilaguamara** [comment]
I have uploaded the changes with a 3.10 running agent and then copied in 3.11, but it seems that there are more things that I had not tried for 3.11. I will solve the changes again so that all SCAs work as planned.

File: debs/SPECS/3.11.0/wazuh-agent/debian/rules [hunk 1]
```diff
@@ -74,6 +74,9 @@ override_dh_install:
 	rm -rf ${TARGET_DIR}$(INSTALLATION_DIR)/ruleset/sca
 	mkdir -p ${TARGET_DIR}$(INSTALLATION_DIR)/ruleset/sca
 
+	# Copying ossec-init.conf
+	${SCRIPTS_DIR}/gen_ossec.sh init agent ${DIR} > ${DIR}/etc/ossec-init.conf
```
Comment (BraulioV) [inline debs/SPECS/3.11.0/wazuh-agent/debian/rules]: This file has already been created by the `install.sh` while it was executed. In v3.10.0 rules files you are copying the file but here you are only creating it but it isn't copied.

File: debs/SPECS/3.9.5/wazuh-agent/debian/rules [hunk 2]
```diff
@@ -23,6 +23,20 @@ export INSTALLATION_DIR="/var/ossec"
 export INSTALLATION_SCRIPTS_DIR="${INSTALLATION_DIR}/packages_files/agent_installation_scripts"
 export JOBS="5"
 export DEBUG_ENABLED="no"
+export OS=$(lsb_release -si)
```
Comment (BraulioV) [inline debs/SPECS/3.9.5/wazuh-agent/debian/rules]: These changes aren't necessary. In addition, you are changing the rules file of v3.9.5 and we won't release a new package of this version.

**Comment by BraulioV** [comment]
Hi @alguacilaguamara,

Can you check the requested changes? In addition, once you made those changes can you perform the following tests?
- [ ] Fresh install wazuh-manager v3.10.0 and check the date of `ossec-init.conf` file.
- [ ] Fresh install wazuh-manager v3.11.0 and check the date of `ossec-init.conf` file.
- [ ] Fresh install wazuh-agent v3.10.0 and check the date of `ossec-init.conf` file.
- [ ] Fresh install wazuh-agent v3.11.0 and check the date of `ossec-init.conf` file.
- [ ] Upgrade a wazuh-manager package v3.9.5 to v3.10.0 and check the date of `ossec-init.conf` file.
- [ ] Upgrade a wazuh-manager package v3.9.5 to v3.11.0 and check the date of `ossec-init.conf` file.
- [ ] Upgrade a wazuh-manager package v3.10.0 to v3.11.0 and check the date of `ossec-init.conf` file.
- [ ] Upgrade a wazuh-agent package v3.9.5 to v3.10.0 and check the date of `ossec-init.conf` file.
- [ ] Upgrade a wazuh-agent package v3.9.5 to v3.11.0 and check the date of `ossec-init.conf` file.
- [ ] Upgrade a wazuh-agent package v3.10.0 to v3.11.0 and check the date of `ossec-init.conf` file.

File: debs/SPECS/3.10.0/wazuh-agent/debian/rules [hunk 3]
```diff
@@ -170,6 +175,8 @@ override_dh_install:
 
 	cp src/systemd/wazuh-agent.service ${TARGET_DIR}/etc/systemd/system/
 
+
```
[the remark this answers is not in the document]
Comment (alguacilaguamara) [reply debs/SPECS/3.10.0/wazuh-agent/debian/rules]: I erase it

File: debs/SPECS/3.10.0/wazuh-manager/debian/postinst [hunk 4]
```diff
@@ -100,9 +96,9 @@ case "$1" in
                 mv $passlist.bak $passlist
             fi
         fi
-    else
-        ${SCRIPTS_DIR}/gen_ossec.sh conf manager ${OS} ${VER} > ${DIR}/etc/ossec.conf.new
-        chmod 660 ${DIR}/etc/ossec.conf.new
+    #else
```
[the remark this answers is not in the document]
Comment (alguacilaguamara) [reply debs/SPECS/3.10.0/wazuh-manager/debian/postinst]: Originally doing a test, I erased it.

File: debs/SPECS/3.11.0/wazuh-agent/debian/rules (hunk 1 above)
Comment (alguacilaguamara) [reply debs/SPECS/3.11.0/wazuh-agent/debian/rules]: Error of previous tests

**Comment by alguacilaguamara** [comment]
Hi @BraulioV .

To do the installation of the packages and the updates I found the following make error.

```
cp: cannot stat `etc/templates/config/darwin/sca.files': No such file or directory
make[1]: *** [override_dh_install] Error 1
make[1]: Leaving directory `/build_wazuh/agent/wazuh-agent-3.10.0'.
make: *** [binary] Error 2
```

To try to force the generation, I have commented 
https://github.com/wazuh/wazuh-packages/blob/40c2f05bae2b73e98b00a7aac7f260011774ba8c/debs/SPECS/3.10.0/wazuh-agent/debian/rules#L135-L138

But the problem remains unresolved.

File: debs/SPECS/3.10.0/wazuh-manager/debian/postinst (hunk 4 above)
[the remark this answers is not in the document]
Comment (alguacilaguamara) [reply debs/SPECS/3.10.0/wazuh-manager/debian/postinst]: Restored those lines of code and removed the ossec-init.conf from the postinst tasks in https://github.com/wazuh/wazuh-packages/pull/291/commits/d03b039d3ae1f707cbf5cf61362bc227f7698ad7

**Comment by alguacilaguamara** [comment]
Hello Braulio,

In the update tests, for example from 3.9.5 to 3.10 of the Wazuh agent I get the following information:

image

Is this a problem or can you keep it that way?

**Pull request closed without merging**